File: db4on/reference/vb/db4o-reference-chapters/src/com/db4odoc/f1/diagnostics/CarEvaluation.vb
' Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com 

Imports com.db4o.query

Namespace com.db4odoc.f1.diagnostics
    Public Class CarEvaluation
        Implements Evaluation
        Public Sub Evaluate(ByVal candidate As Candidate) Implements Evaluation.Evaluate
            Dim car As evaluations.Car = CType(candidate.GetObject(), evaluations.Car)
            candidate.Include(car.Model.EndsWith("2002"))
        End Sub

    End Class
End Namespace

File: db4on/reference/db4o-reference-chapters/src/com/db4odoc/f1/diagnostics/CarEvaluation.cs
/* Copyright (C) 2004 - 2006 db4objects Inc. http://www.db4o.com */

using com.db4o.query;

namespace com.db4odoc.f1.diagnostics
{

	public class CarEvaluation:Evaluation 
	{
		public void Evaluate(Candidate candidate)
		{
			Car car=(Car)candidate.GetObject();
			candidate.Include(car.Model.EndsWith("2002"));
		}
	}
}